distributed/client.py

```diff
@@ -3390,6 +3390,26 @@ def get_task_stream(self, start=None, stop=None, count=None):
         return self.sync(self.scheduler.get_task_stream, start=start,
                          stop=stop, count=count)
 
+    @gen.coroutine
+    def _register_init_func(self, function):
+        response = yield self.scheduler.register_init_func(function=dumps(function))
+        raise gen.Return(response)
+
+    def register_init_func(self, function, asynchronous=None):
+        """
+        Registers and run an initialization for every worker
+
+        This registers a function to be called by every currently connected
+        workers, and that will be also called by any added worker after that.
+        Multiple functions can be added by multiple calls.
+
+        Parameters
+        ----------
+        function: Function to register and run
+        """
+        return self.sync(self._register_init_func, function,
+                         asynchronous=asynchronous)
```

Comment:
This logic is simple enough that we probably don't need a separate coroutine, but can instead probably call the scheduler method directly here:

    return self.sync(self.scheduler.register_init_func, ...

You can then drop the `_register_init_func` method entirely.

Reply:
Done too.